modules/programs/neovim.nix

```diff
@@ -232,6 +248,20 @@ in {
       coc = {
         enable = mkEnableOption "Coc";
 
+        plugins = mkOption {
```

Comment:
The handling of neovim regarding packpath/runtimepath has evolved so fast I dont know what the current state is but there must be a better way than distinguishing coc plugins from regular plugins.

Reply:
> there must be a better way than distinguishing coc plugins from regular plugins.

I don't even think this is a bad thing? They're generally not actually vim plugins... I'm not even sure if there even exist any coc plugins that are polyglots (are any actually also vim plugins?), and therefore they likely don't need to be in the `packpath` (nor even `runtimepath`); ideally coc would just have a separate variable/configuration option for telling it where to look for them?

[coc.nvim has its own explicit logic for looking in runtimepath](https://github.com/neoclide/coc.nvim/blob/0219acc58d6fd43460711e80bb1ab147f76eb5eb/src/extensions.ts#L680), which is how this worked previously, where I assume h-m used runtimepath instead of packpath. The logic certainly could be patched/improved, but as it stands `runtimepath` is currently the only way it knows how to load external extensions (outside of the [single-file plugins](https://github.com/neoclide/coc.nvim/wiki/Using-coc-extensions#single-file-extensions)). Perhaps we could simulate a `:CocInstall` instead; I'm not sure where it puts those?

Reply:
I think it's a good thing to keep the coc-plugins separate from other plugins. The coc plugins are fundamentally not vim plugins (they are written in JS, and generally interface with CoC rather than with neovim.